File: app/src/main/java/com/wiser/rollnumber/RunTextView.java
package com.wiser.rollnumber;

import android.animation.ObjectAnimator;
import android.content.Context;
import android.util.AttributeSet;
import android.view.animation.AccelerateDecelerateInterpolator;

import androidx.appcompat.widget.AppCompatTextView;

public class RunTextView extends AppCompatTextView {
	
	private int duration = 1500;
	private float number;
 
	public float getNumber() {
		return number;
	}
 
	public void setNumber(float number) {
		this.number = number;
		setText(String.format("%,.2f",number));
		
	}
 
	public RunTextView(Context context, AttributeSet attrs) {
		super(context, attrs);
	}
	
	/**
	 * 显示
	 * @param number
	 */
	public void runWithAnimation(float number){
		ObjectAnimator objectAnimator = ObjectAnimator.ofFloat(
				this, "number", 0, number);
		objectAnimator.setDuration(duration);
		objectAnimator.setInterpolator(new AccelerateDecelerateInterpolator());
		
		objectAnimator.start();
		
	}
}

File: roll/src/main/java/com/wiser/roll/RunTextView.kt
package com.wiser.roll

import android.animation.ObjectAnimator
import android.content.Context
import android.util.AttributeSet
import android.view.animation.AccelerateDecelerateInterpolator
import androidx.appcompat.widget.AppCompatTextView

/**
 * @author Wiser
 */
class RunTextView(context: Context, attrs: AttributeSet) : AppCompatTextView(context, attrs) {

    private var duration = 1500

    private var isAutoStart = false

    var number = 0f
        set(number) {
            field = number
            text = String.format("%,.2f", number)
        }

    init {
        val ta = context.obtainStyledAttributes(attrs,R.styleable.RunTextView)
        duration = ta.getInt(R.styleable.RunTextView_rtv_duration,duration)
        number = ta.getFloat(R.styleable.RunTextView_rtv_number,number)
        isAutoStart = ta.getBoolean(R.styleable.RunTextView_rtv_auto,isAutoStart)
        ta.recycle()

        if (isAutoStart) {
            startAnimator(number)
        }
    }

    fun setDuration(duration: Int): RunTextView? {
        this.duration = duration
        return this
    }

    /**
     * 显示
     * @param number
     */
    fun startAnimator(number: Float) {
        val objectAnimator = ObjectAnimator.ofFloat(
            this, "number", 0f, number
        )
        objectAnimator.duration = duration.toLong()
        objectAnimator.interpolator = AccelerateDecelerateInterpolator()
        objectAnimator.start()
    }
}